hare/algorithm.go

```diff
@@ -66,32 +66,32 @@ type ConsensusProcess struct {
 	terminating       bool
 	cfg               config.Config
 	notifySent        bool
-	pending           map[string]Message
+	pending           map[string]*pb.HareMessage
 }
 
 func NewConsensusProcess(cfg config.Config, instanceId InstanceId, s *Set, oracle Rolacle, signing Signing, p2p NetworkService, terminationReport chan TerminationOutput, logger log.Log) *ConsensusProcess {
 	proc := &ConsensusProcess{}
 	proc.State = State{-1, -1, s.Clone(), nil}
 	proc.Closer = NewCloser()
 	proc.instanceId = instanceId
-	proc.oracle = oracle
+	proc.oracle = newHareOracle(oracle, cfg.N)
 	proc.signing = signing
 	proc.network = p2p
-	proc.validator = NewMessageValidator(signing, cfg.F+1, cfg.N, proc.statusValidator(), logger)
+	proc.validator = newSyntaxContextValidator(signing, cfg.F+1, proc.statusValidator(), logger)
 	proc.preRoundTracker = NewPreRoundTracker(cfg.F+1, cfg.N)
 	proc.notifyTracker = NewNotifyTracker(cfg.N)
 	proc.terminating = false
 	proc.cfg = cfg
 	proc.notifySent = false
 	proc.terminationReport = terminationReport
-	proc.pending = make(map[string]Message, cfg.N)
+	proc.pending = make(map[string]*pb.HareMessage, cfg.N)
 	proc.Log = logger
 
 	return proc
 }
 
 func (proc *ConsensusProcess) Id() uint32 {
```

Comment:
why do you need to work with int32?

Reply:
Same reason as before, we use the interface Identifiable, in this case, in the broker.

Reply:
Fixed